File: Frontend/src/app/Models/booking-dto.ts
export interface BookingDTO {
  UnitId: number;
  CheckInDate: Date;
  CheckOutDate: Date;
  NumberOfGuests: number;
}

export interface BookingResponseDTO {
  id: number;
  unitId: number;
  tenantId: string;
  checkInDate: Date;
  checkOutDate: Date;
  numberOfGuests: number;
  bookingDate: Date;
  totalPrice: number;
  ownerPayoutAmount: number;
  platformComission: number;
  paymentStatus: string;
  unitReviewed: boolean;
  ownerReviewd: boolean;
  tenantName?: string; // Make sure this line is present and uncommented
  qrCodeUrl?: string;

  // Unit details (if populated)
  unit?: {
    id: number;
    title: string;
    address: string;
    villageName: string;
    unitType: string;
    basePricePerNight: number;
    ownerName: string;
  };
}

File: Backend/API/DTOs/BookingDTOs/BookingDTO.cs
using API.Models;
using System.ComponentModel.DataAnnotations.Schema;

using API.DTOs.UnitDTO;

namespace API.DTOs.BookingDTOs
{
    public class BookingDTO
    {
        //from booking
        public int Id { get; set; }
        public int UnitId { get; set; }
        public DateTime CheckInDate { get; set; }
        public DateTime CheckOutDate { get; set; }
        public int NumberOfGuests { get; set; }
        public string TenantId { get; set; }
        public DateTime BookingDate { get; set; }


        public decimal TotalPrice { get; set; }
        public decimal OwnerPayoutAmount { get; set; }
        public decimal PlatformComission { get; set; }
        public string PaymentStatus { get; set; }
        public bool unitReviewed { get; set; }
        public UnitDetailsDTO Unit{ get; set; }



    }
}
